update/stock_update.js

```diff
@@ -1,122 +1,128 @@
-//---CONFIGURE AUTHENTICATION HERE--------------------
-var CLOUDANT_KEY      = 'XXXXXXXXX';
-var CLOUDANT_PASSWORD = 'YYYYYYYYY';
-var CLOUDANT_ACCESS   = 'someurl-bluemix.cloudant.com'
-var DB_NAME = 'DB_NAME';
-//----------------------------------------------------
 
-var Cloudant = require('cloudant');
-
-var cloudant = Cloudant({
-    account: CLOUDANT_ACCESS,
-    key: CLOUDANT_KEY,
-    password: CLOUDANT_PASSWORD
-});
-var db = cloudant.db.use(DB_NAME);
+const stock_db = require('./stock_db.js');
+const utils = require('./utils.js');
+const discovery = require('./discovery.js');
 
 //TODO these should be the companies' tickers
 var companies = ['A', 'B', 'C', 'D'];
 
-function isFunc(func) {
-    return typeof func == 'function';
-}
-
-function getDocs(callback) {
-    db.list({
-        //this field is needed to return all doc data
-        include_docs: true
-    }, function(err, data) {
-        if (isFunc(callback)) {
-            var docs = data.rows.map(function(row) {
-                return row.doc;
-            });
-            callback(err, docs);
-        }
-    });
-}
-
-//include _id and _rev of existing doc in the doc to perform an update
-function insertOrUpdateDoc(doc, callback) {
-
-    db.insert(doc, function(err, body, header) {
-        if (err) {
-            console.log('insertion failed: ' + err.message);
-        } else if (isFunc(callback)) {
-            callback(body, header);
-        }
-    });
-}
-
 function findStockDatum(stocks, company) {
 
-    for (var i = 0; i < stocks.length; i++) {
-        var stock = stocks[i];
-        if (stock.ticker == company) {
-            return stock;
-        }
+  for (var i = 0; i < stocks.length; i++) {
+    var stock = stocks[i];
+    if (stock.ticker == company) {
```

Comment:
consider changing both `stock.ticker` and `company` to lower case and then doing a comparison to "IBM" and "ibm" will still match. see https://stackoverflow.com/questions/154862/convert-javascript-string-to-be-all-lower-case

Reply:
good point